src/core/Akka/Dispatch/CachingConfig.cs

```diff
@@ -0,0 +1,277 @@
+﻿/**
+ * Copyright (C) 2009-2015 Typesafe Inc. <http://www.typesafe.com>
+ * Original C# code written by Akka.NET project <http://getakka.net/>
+ */
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using Akka.Configuration;
+using Akka.Configuration.Hocon;
+
+namespace Akka.Dispatch
+{
+    /// <summary>
+    /// INTERNAL API
+    /// 
+    /// A <see cref="CachingConfig"/> is a <see cref="Config"/> that wraps another <see cref="Config"/> and is used to
+    /// cache path lookup and string retrieval, which we happen to do in some critical paths of the actor creation
+    /// and mailbox selection code.
+    /// 
+    /// All other <see cref="Config"/> operations are delegated to the wrapped <see cref="Config"/>.
+    /// </summary>
+    class CachingConfig : Config
```

Comment:
We are also passing bits of configurations across the wire, e.g. some deployment primitives holds a `Config` objects.
This _should_ be handled by the surrogate stuff, but I guess we need to make sure?